internal/utils/config.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import (
+	"io/ioutil"
+	"os"
+)
+
+// Configuration holds the configuration to connect to the Argo CD server
+type Configuration struct {
+	ArgoCDAuthToken  string
+	ArgoCDServerAddr string
+}
+
+// readFromFile returns the content of a file
+func readFromFile(path string) (string, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+
+	defer file.Close()
+
+	data, err := ioutil.ReadAll(file)
+	if err != nil {
+		return "", err
+	}
+
+	return string(data), nil
+}
+
+// readFromEnvOrFile returns the value of an environment variable if present, or from a file otherwise
+func readFromEnvOrFile(secretName string) (string, error) {
+	var err error
+	value := os.Getenv(secretName)
+
+	if len(value) == 0 {
+		value, err = readFromFile(ConfigDirectory + secretName)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return value, nil
+}
+
+
+/*
+ReadConfiguration returns a Configuration, reading it from environment variables if present, or from files in
+the configuration directory otherwise
+*/
+func ReadConfiguration() (*Configuration, error) {
```

Comment:
why using pointer semantics here?
we're using value semantics everywhere else?

Reply:
it's a minor thing, tbh, the logic of using pointer semantics makes sense so we can use `nil`. It's just against the decision made in #33.

@maruina what do you think?

Reply:
Uhm - we can go with value semantics and return an empty config. As long as it also returns an error, we're able to handle it.

Reply:
Depending how we make things complicated, but it would be good to follow the same convention.